cmd/internal/register/maps.go

```diff
@@ -59,34 +63,24 @@ func (e ErrDefaultTagsInvalid) Error() string {
 }
 
 func AutoConfigMapLayers(providers map[string]provider.Tiler) []config.MapLayer {
-	mapLayers := []config.MapLayer{}
-	for pname, p := range providers {
-		// log.Println(p.String("name", nil))
-		provLayers, err := p.Layers()
-		if err != nil {
-			log.Println(err)
-		}
-
-		for _, l := range provLayers {
-			providerLayer := fmt.Sprintf("%v.%v", pname, l.Name())
-			mapLayers = append(mapLayers, config.MapLayer{
-				ProviderLayer: env.String(providerLayer),
-				Name:          env.String(l.Name()),
-			})
+	mapLayers := make([]config.MapLayer, len(providers))
```

Comment:
Looking at this a bit more, @stdmn was grabbing the providerLayers and @gdey you suggested he remove them. Don't we need them?

Reply:
@gdey @ARolek Once I have some more clarity on this, I can update accordingly.

Reply:
@ARolek No, because we have made it so that "providerName.*" will grab all the layers. No need to iterate the layers here. 

We are going to be generating the same number of MapLayers as there are providers. Therefor it make sense to preallocate the entries. However, since we are iterating a map, we don't get an index, so we have to use a `idx := 0` to track the index. Renaming `count` to `idx` is probally the right move here.  

The one thing I think I would add; is that we sort the names of the providers, so that when we use this code to write out config files later, the names will be in order:

```go

mapLayers := make([]config.MapLayers, len(providers))
names := make([]string,len(providers))

idx := 0
for pname, _ := range providers {
         names[idx] = pname
         idx++
}
sort.Strings(names)

for idx := range names {
       mapLayers[idx] = config.MapLayer{
                 ProviderLayer: env.String(fmt.Sprintf("%s.*",names[idx])),
         }
}

return mapLayers
```

Reply:
@gdey I see. I missed that detail so I think the fact that it's so subtle but yet powerful it should be documented in code. 

One advantage of outputting all the layer names is that the user could then modify a config file to adjust the zoom ranges for the various layers. With the `*` approach it's more cumbersome for subsequent refinement of the auto-generated config file.

Reply:
Listing all the layers has a subtly different meaning than `*`. List all the layers says that these layers and only these layers will be configured, even if the layers change; whereas `*` will check the set of layers every time. In this case I think `*` is more apporiate. For Configuration generation, it is will become important for the config generation tool to be able to handle both cases, but I think the default should be `*`. These concerns should not bog this pr down. And should be added to an issue for config generation.

Reply:
That's a very big difference. So when `*` is used the layer list is fetched for every request rather than building the layer list at config parse time. This needs to be documented in code better as it's not clear but I see the reasoning now.

Reply:
Also, using '*' here removes on extra loop. Since we always loop through all the provider layers later on, when building out the full config.

Reply:
Not, on every request just on config parse time. Sorry for the confusion. Which is after every restart of the app. Whereas before it's always the same. Here functionally it's the same (except this `*` is faster because it's not iterating the layers again.) startup wise. I was talking about config generation.